source/systems.lua

```diff
@@ -2,20 +2,12 @@ local systems = {}
 systems.update = {}
 systems.draw = {}
 
-local systemsMetatable = {}
-
-local function addUpdateFunction(callback)
-    table.insert(systems.update, callback)
-end
-
-local function addDrawFunction(callback, priority)
+function systems.addDrawFunction(callback, priority)
     systems.draw[priority] = callback
```

Comment:
This only allows you to have a single function at each priority with no overlap. Instead you should iterate over all draw functions and slot it in based on the priority.